src/app/components/champion/ChampionSelector.tsx

```diff
@@ -1,46 +1,66 @@
-import React, {useCallback, useEffect, useState} from "react";
+import React, {useCallback, useState} from "react";
+import axios from 'axios';
 import {Champion, Data} from "../model/models";
 import {ChampionView} from "./ChampionView";
-import {getUrlChampion, getUrlChampionList} from "../../config/config";
-import axios from 'axios';
-import {defaultChampion} from "./default";
-
+import {getUrlChampion, getUrlChampionAvatar, getUrlChampionList} from "../../config/config";
+import {useFetchAPI} from "../../api/reducer";
+import {Autocomplete} from "@material-ui/lab";
+import {Avatar, List, TextField} from "@material-ui/core";
+import {makeStyles} from "@material-ui/core/styles";
 
 export const ChampionSelector = () => {
-    const [champion, setChampion] = useState<Champion>(defaultChampion);
-    const [championsList, setChampionsList] = useState<Champion[]>([]);
+    const [championId, setChampionId] = useState("Aatrox")
+    const championListState = useFetchAPI<Data>({data: {}}, () => axios.get(getUrlChampionList()))
+    const championState = useFetchAPI<Data>(undefined, () => axios.get(getUrlChampion(championId)), [championId])
 
-    useEffect(() => {
-        fetchChampionList()
-    }, [])
-
-    function fetchChampionList() {
-        axios.get(getUrlChampionList(), {}).then(response => {
-            const payload = response.data as Data
-            const champs = Object.values(payload.data) as Champion[]
-            setChampionsList(champs)
-        })
+    function extractChampionList(data: Data) {
+        return Object.values(data.data) as Champion[]
     }
 
-    function loadChampion(championSelected:string) {
-        axios.get(getUrlChampion(championSelected), {}).then(response => {
-            const payload = response.data as Data
-            const champs = Object.values(payload.data) as Champion[]
-            const champion = champs[0]
-            setChampion(champion as Champion)
-        })
+    function extractChampion(data: Data) {
+        return Object.values(data.data)[0] as Champion
```

Comment:
That's exactly the problem of using Data ! You use the same interface to treat 2 differents format of the payload. (champion list and champion). Those methods are not bad at all. The use of type Data is.